test/SlowTests/Server/Documents/ETL/Elasticsearch/ElasticsearchEtlTests.cs

```diff
@@ -0,0 +1,871 @@
+                            new OrderLine {Cost = 4, Product = "Bear", Quantity = 2},
+                        }
+                    });
+                    await session.SaveChangesAsync();
+                }
+
+                var result1 = store.Maintenance.Send(new PutConnectionStringOperation<ElasticsearchConnectionString>(new ElasticsearchConnectionString()
+                {
+                    Name = "simulate", Nodes = new[] {"http://localhost:9200"}
+                }));
+                Assert.NotNull(result1.RaftCommandIndex);
+
+                var database = GetDatabase(store.Database).Result;
+
+                using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                {
+                    var result = (ElasticsearchEtlTestScriptResult)ElasticsearchEtl.TestScript(
+                        new TestElasticsearchEtlScript
+                        {
+                            DocumentId = "orders/1-A",
+                            IsDelete = true,
+                            Configuration = new ElasticsearchEtlConfiguration()
+                            {
+                                Name = "simulate",
+                                ConnectionStringName = "simulate",
+                                ElasticIndexes =
+                                {
+                                    new ElasticsearchIndex {IndexName = "Orders", IndexIdProperty = "Id"},
+                                    new ElasticsearchIndex {IndexName = "OrderLines", IndexIdProperty = "OrderId"},
+                                    new ElasticsearchIndex {IndexName = "NotUsedInScript", IndexIdProperty = "OrderId"},
+                                },
+                                Transforms =
+                                {
+                                    new Transformation() {Collections = {"Orders"}, Name = "OrdersAndLines", Script = defaultScript + "output('test output')"}
+                                }
+                            }
+                        }, database, database.ServerStore, context);
+
+                    Assert.Equal(0, result.TransformationErrors.Count);
+
+                    Assert.Equal(2, result.Summary.Count);
+
+                    var orderLines = result.Summary.First(x => x.IndexName == "orderlines");
+
+                    Assert.Equal(1, orderLines.Commands.Length); // delete
+
+                    var orders = result.Summary.First(x => x.IndexName == "orders");
+
+                    Assert.Equal(1, orders.Commands.Length); // delete
+                }
+                
+                using (var session = store.OpenAsyncSession())
+                {
+                    Assert.NotNull(session.Query<Order>("orders/1-A"));
+                }
+            }
+        }
+
+        protected void SetupElasticEtl(DocumentStore store, string script, IEnumerable<string> collections = null, bool applyToAllDocuments = false)
```

Comment:
How do we cleanup Elastic between calls to the tests?

Reply:
We can delete indexes after every test. Is that sound ok? @ayende

Reply:
Yes, that should do it.